pkg/gardenlet/controller/managedseed/managedseed_reconciler.go

```diff
@@ -0,0 +1,280 @@
+	if err := a.CreateOrUpdate(r.ctx, managedSeed, shoot); err != nil {
+		managedSeedLogger.Errorf("Could not reconcile ManagedSeed creation or update: %+v", err)
+
+		// Record an event
+		lastError := &gardencorev1beta1.LastError{
+			Codes:       gardencorev1beta1helper.ExtractErrorCodes(err),
+			Description: "Could not reconcile creation or update: " + err.Error(),
+		}
+		r.recorder.Eventf(managedSeed, corev1.EventTypeWarning, gardencorev1beta1.EventReconcileError, "%s", lastError.Description)
+
+		// Update status to Error
+		if updateErr := updateStatusError(r.ctx, gardenClient.DirectClient(), managedSeed, lastError.Description, lastError); updateErr != nil {
+			managedSeedLogger.Errorf("Could not update ManagedSeed status after creation or update error: %+v", updateErr)
+			return reconcile.Result{}, updateErr
+		}
+
+		// Return error result
+		return reconcile.Result{}, errors.New(lastError.Description)
+	}
+	managedSeedLogger.Debug("Successfully reconciled ManagedSeed creation or update")
+
+	// Update status to Succeeded
+	if updateErr := updateStatusSucceeded(r.ctx, gardenClient.DirectClient(), managedSeed, "Reconciled successfully"); updateErr != nil {
+		managedSeedLogger.Errorf("Could not update ManagedSeed status after creation or update success: %+v", updateErr)
+		return reconcile.Result{}, updateErr
+	}
+
+	// Return success result
+	return reconcile.Result{}, nil
+}
+
+func (r *reconciler) delete(gardenClient kubernetes.Interface, managedSeed *seedmanagementv1alpha1.ManagedSeed) (reconcile.Result, error) {
+	managedSeedLogger := logger.NewFieldLogger(r.logger, "managedSeed", kutil.ObjectName(managedSeed))
+
+	// Check gardener finalizer
+	if !sets.NewString(managedSeed.Finalizers...).Has(gardencorev1beta1.GardenerName) {
+		managedSeedLogger.Debug("Skipping ManagedSeed as it does not have a finalizer")
+		return reconcile.Result{}, nil
+	}
+
+	// Get shoot
+	shoot := &gardencorev1beta1.Shoot{}
+	if err := gardenClient.Client().Get(r.ctx, kutil.Key(managedSeed.Namespace, managedSeed.Spec.Shoot.Name), shoot); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Get seed client
+	seedClient, err := r.clientMap.GetClient(r.ctx, keys.ForSeedWithName(*shoot.Spec.SeedName))
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("could not get seed client: %w", err)
+	}
+
+	// Get shoot client
+	shootClient, err := r.clientMap.GetClient(r.ctx, keys.ForShoot(shoot))
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("could not get shoot client: %w", err)
+	}
+
+	// Create actuator
+	a := newActuator(gardenClient, seedClient, shootClient, r.config, r.imageVector, r.logger)
+
+	// Update status to Processing
+	if updateErr := updateStatusProcessing(r.ctx, gardenClient.DirectClient(), managedSeed, "Deleting"); updateErr != nil {
+		managedSeedLogger.Errorf("Could not update ManagedSeed status before deletion: %+v", updateErr)
+		return reconcile.Result{}, updateErr
+	}
+
+	// Reconcile deletion
```

Comment:
This is in fact really "reconcile deletion", not "execute deletion". The actuator `Delete` method really performs reconciliation, not deletion, see my comments above.

Reply:
Yeah, it's just that we never used it this way. We have "reconciliation" (create or update) and "deletion".

Reply:
I still find this inaccurate and confusing and would like to improve this going forward. I think `Execute deletion` is simply wrong here, and I would like to keep `reconcile deletion` here and elsewhere in this file (mostly in comments, also in a few messages). Since this is not an API / method name, I don't think consistency is that important, and hopefully we could change this in APIs / method names as well at some point (see my earlier comments)